euphoria-service/src/main/java/edu/cooper/ece366/euphoria/store/jdbc/DataSource.java

```diff
@@ -0,0 +1,24 @@
+package edu.cooper.ece366.euphoria.store.jdbc;
+
+import org.apache.commons.dbcp.BasicDataSource;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+// DataSource class adapted from https://stackoverflow.com/a/7592081
+public final class DataSource {
+    private static final BasicDataSource ds = new BasicDataSource();
+
+    static {
```

Comment:
not ideal to have to hard code here (instead of in `.conf` file), but this should work. if you test your store classes, they will be tightly coupled to needing there to be this specific sql instance running (otherwise those tests would fail). 

you could instead initialize `DataSource` (not statically) in your main with the rest of your dependencies based on the `environment.config()`. then testing it will be easier.

but your group is also a bit ahead in implementing connection pooling anyway, so don't over-optimize against this. just a suggestion

Reply:
Implemented in f95409a